crates/pallet-domains/src/lib.rs

```diff
@@ -1908,9 +1913,10 @@ mod pallet {
         }
 
         fn on_finalize(_: BlockNumberFor<T>) {
-            // If this consensus block will derive any domain block, gather the necessary storage for potential fraud proof usage
+            // If this consensus block will derive any domain block, gather the necessary storage
+            // for potential fraud proof usage
             if SuccessfulBundles::<T>::iter_keys().count() > 0
-                || DomainRuntimeUpgrades::<T>::exists()
+                || !DomainRuntimeUpgrades::<T>::get().is_empty()
```

Comment:
Is this the check we want?

Now that we're always populating `DomainRuntimeUpgrades`, `DomainRuntimeUpgrades::<T>::exists()` would always be true, so the `InvalidInherentExtrinsicData` will be created for each block. Instead, this checks if there were any upgrades.

Reply:
Yeah, this check makes sense to me.